### DISCUSSION TITLE
Suddenly not working anymore (after recreation)

### DISCUSSION THREAD

**Discussion by the-hotmann** [haugene/docker-transmission-openvpn#1952] [Q&A]
### Is there a pinned issue for this?

- [X] I have read the pinned issues and could not find my issue

### Is there an existing or similar issue/discussion for this?

- [X] I have searched the existing issues
- [X] I have searched the existing discussions

### Is there any comment in the documentation for this?

- [X] I have read the documentation, especially the FAQ and Troubleshooting parts

### Is this related to a provider?

- [X] I have checked the provider repo for issues
- [X] My issue is NOT related to a provider

### Are you using the latest release?

- [X] I am using the latest release

### Have you tried using the dev branch latest?

- [X] I have tried using dev branch

### Config used

version: '3.3'
services:
    transmission-openvpn:
        cap_add:
            - NET_ADMIN
        volumes:
            - '[REDACTED]:/config'
            - '[REDACTED]:/data'
        environment:
            - OPENVPN_PROVIDER=goosevpn
            - OPENVPN_CONFIG=[REDACTED]
            - OPENVPN_USERNAME=[REDACTED]
            - OPENVPN_PASSWORD=[REDACTED]
            - LOCAL_NETWORK=172.17.0.0/24
            - TZ=Europe/Berlin
            - WEBPROXY_ENABLED=true
            - WEBPROXY_PORT=8118
            - PUID=10003
            - PGID=1003
        logging:
            driver: json-file
            options:
                max-size: 10m
        ports:
            - '8118:8118'
            - '9091:9091'
        dns:
            - 1.1.1.1
            - 1.0.0.1
        image: haugene/transmission-openvpn:dev
        container_name: transmission-openvpn
        restart: always
        privileged: true

### Current Behavior

When wanting to run:
`docker run -d --link transmission-openvpn:transmission -p 8080:8080 haugene/transmission-openvpn-proxy`
it throws this error:
`docker: Error response from daemon: Cannot link to /transmission-openvpn, as it does not belong to the default network.`

### Expected Behavior

Linking containers as usual. As I have not changed my settings.

### How have you tried to solve the problem?

Well, from restarting the servers to prune the docker networks, but nothing helped.

### Log output

proxy reportes:

`{"message":"Cannot link to /transmission-openvpn, as it does not belong to the default network"}`

### Environment

```markdown
- OS: Debian 10.10
- Docker: 20.10.3, build 48d30b5
```

### Anything else?

-

**Comment by pkishino** [comment]
This is a docker issue, not a container issue.. check and verify your docker networks. Have you recently updated docker version?
Check that your are using the same network for both containers you want to link

**Reply by the-hotmann** [reply]
Thanks for the reply!
Is there a good example on how to do this for both containers?

I set my container up with "docker-compose" and a yml file to have it more automized/scriptable when I need. I could not read anything usefull in the docs about how to set up the same network for both. And if so, which one.

I usualy take "docker_default", is that correct? Thanks!

**Reply by pkishino** [reply]
If using compose, for every container it will create diff networks.
I suggest creating a named network for this and using that.
This is from my compose, first part is part of this container, second is a main block defining the network
```yaml
        ports:
            - '9091:9091'
        image: haugene/transmission-openvpn:dev
        restart: unless-stopped
        networks:
          transmission:
            ipv4_address: 172.20.10.10


networks:
  transmission:
    name: transmission
    driver: bridge
    ipam:
      config:
        - subnet: "172.20.10.0/24"
```

**Reply by the-hotmann** [reply]
Thanks man, will try it!

**Reply by the-hotmann** [reply]
I just tried it it with this, then I link the (not already created) proxy to it like this:

`docker run -d --link transmission-openvpn:transmission --net transmission -p 8080:8080 haugene/transmission-openvpn-proxy`

which will create a `haugene/transmission-openvpn-proxy` instance linking it to the container I have chosen. After this I get a `502 Bad Gateway nginx/1.21.3`.
Logs say:

`upstream, client: 123.123.123.123, server: , request: "GET /favicon.ico HTTP/1.1", upstream: "http://172.20.10.10:9091/favicon.ico", host: "host.domain.tld:8080", referrer: "http://host.domain.tld:8080/transmission/web/"`

Is there anything you know how to solve this?

**Reply by pkishino** [reply]
hmm, when you get the 502 error, how are you trying to access the UI?
